src/neo4j_genai/core/graph.py

```diff
@@ -0,0 +1,104 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Basic graph structure for Pipeline
+"""
+
+from __future__ import annotations
+
+from typing import Any, Optional
+
+
+class Node:
```

Comment:
I wonder if it might make sense to rename this file and the classes within it. People might interpret `Node`, `Graph`, etc. as Neo4j objects, rather than as components of the KG builder pipeline.

Reply:
I agree it can be confusing, even though it is not meant to be exposed to users. What do you think about the following:
- Graph => ComputationalGraph
- Node => Vertex
- Edge => Edge (unchanged, it is not used in the Neo4j ecosystem)

I'd like to keep them as generic as possible because these classes only contain the graph structure, nothing related to pipeline yet.

Reply:
It might be better if we name them in such a way that it makes it clear they're part of the KG builder pipeline, something like `Graph` -> `KGBuilderGraph`. Although renaming the file or the surrounding folder might be enough to avoid this (maybe `core` to `kg_pipeline`?). I worry imports like `from neo4j_genai.core.graph import Graph` will make people think the `Graph` object represents a Neo4j graph in some way, rather than a computational graph for building knowledge graphs. @willtai do you have any suggestions for this?

Reply:
You convinced me :) Done in: https://github.com/neo4j/neo4j-genai-python/pull/81/commits/7153978424684d8a78bbd4c5b84352ac0a59e3c1